src/hooks/useIntersectionObserver/useIntersectionObserver.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { RefObject, useCallback, useEffect, useRef } from 'react';
+
+export const useIntersectionObserver = (
+	containerRef: RefObject<Element | Document> | undefined,
+	onIntersect: (entry: IntersectionObserverEntry) => void,
+	{ rootMargin, threshold }: Omit<IntersectionObserverInit, 'root'> = {}
+): {
+	observe: (element: Element) => void;
+	unobserve: (element: Element) => void;
+} => {
+	const observerRef = useRef<IntersectionObserver | null>(null);
+	const observedElementsRef = useRef<Element[]>([]);
+
+	const onIntersectRef = useRef<typeof onIntersect>(onIntersect);
+
+	useEffect(() => {
+		onIntersectRef.current = onIntersect;
+	}, [onIntersect]);
+
+	useEffect(() => {
+		if (!observerRef.current && !!containerRef?.current) {
```

Comment:
!observerRef.current -> is this required? considering that the clean function set it to null

Reply:
You're right, I'll remove it